### PR TITLE
[ci] refactor ci bash script

### PR DESCRIPTION
**Pull request by aslonnie**
no conversion result change; just better bash scripting.

### CODE REVIEW

**Comment by aslonnie** [comment]
need to refactor this file so that I can add fake readme in `maketmpl` testdata for unit testing..

File: ci/auto-generate-readme.sh [hunk 1]
```diff
@@ -1,45 +1,63 @@
...
-    fi
-    if [ "$notebook_file" != "../templates/templates/e2e-llm-workflows/README.ipynb" ]; then
-        # Convert notebook file to README.md using nbconvert
-        jupyter nbconvert --to markdown "$notebook_file" --output-dir "$(dirname "$notebook_file")"
-    else
-        echo "Skipping README generation for $notebook_file"
-    fi
-done
+# Search for notebook files named README.ipynb in the ../templates directory
+TEMPLATES_DIRS=($(find "templates" -mindepth 1 -maxdepth 1 -type d))
```
Comment (khluu) [inline ci/auto-generate-readme.sh:14]: just wondering why do we need another pair of parentheses here?

Comment (khluu) [inline ci/auto-generate-readme.sh:14]: just wondering why do we need another pair of parentheses here?

File: ci/auto-generate-readme.sh [hunk 2]
```diff
@@ -1,45 +1,63 @@
...
+	if [[ ! -f "${TMPL}/README.ipynb" ]]; then
+		echo "README.ipynb file not found; skipping notebook conversion and checking."
+	else
+        # Exclude specific notebooks from conversion
+        TMPL_NAME="$(basename "${TMPL}")"
+        NOTEBOOK_FILE="${TMPL}/README.ipynb"
+
+        if [[ "${TMPL_NAME}" == "getting-started" || "${TMPL_NAME}" == "e2e-llm-workflows" || "${TMPL_NAME}" == "ray-summit-multi-modal-search" ]]; then
+            echo "Skip 'Time to complete' checking for ${TMPL_NAME}"
+        elif ! grep -q "Time to complete" "${NOTEBOOK_FILE}" ; then
```
Comment (khluu) [inline ci/auto-generate-readme.sh:32]: maybe double bracket here

**Review by khluu — APPROVED** [review]
LGTM syntax wise!

File: ci/auto-generate-readme.sh (hunk 1 above)
Comment (aslonnie) [reply ci/auto-generate-readme.sh:14]: the inner one is capturing the output, the outer one converts it into a bash array.

Comment (khluu) [reply ci/auto-generate-readme.sh:14]: oh got it

**Pull request merged**